File: backend_spring/src/main/java/com/buddy/pium/dto/post/PostResponse.java
// PostResponse.java
package com.buddy.pium.dto.post;

import com.buddy.pium.entity.post.Post;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class PostResponse {
    private Long id;
    private String title;
    private String content;
    private String category;
    private String imgUrl;
    private String author;
    private Long viewCount;
    private String createdAt;
    private Integer likeCount;
    private Boolean isLiked;
    private Integer commentCount;

    public static PostResponse from(Post post, Long memberId) {
        return PostResponse.builder()
                .id(post.getId())
                .title(post.getTitle())
                .content(post.getContent())
                .category(post.getCategory())
                .imgUrl(post.getImgUrl())
                .author(post.getMember().getNickname())
                .viewCount(post.getViewCount())
                .createdAt(post.getCreatedAt().toString())
                .likeCount(post.getLikes().size())
                .isLiked(post.isLikedBy(memberId))
                .commentCount(post.getComments().size())
                .build();
    }
}

File: backend_spring/src/main/frontend_flutter/lib/models/post_response.dart
class PostResponse {
  final int id;
  final String title;
  final String content;
  final String category;
  final String? postImg;
  final String writer;
  final int viewCount;
  final String createdAt;

  PostResponse({
    required this.id,
    required this.title,
    required this.content,
    required this.category,
    this.postImg,
    required this.writer,
    required this.viewCount,
    required this.createdAt,
  });

  factory PostResponse.fromJson(Map<String, dynamic> json) {
    return PostResponse(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      category: json['category'],
      postImg: json['postImg'],
      writer: json['writer'],
      viewCount: json['viewCount'],
      createdAt: json['createdAt'],
    );
  }
}
